src/restic_fmt_plug.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * This software is Copyright (c) 2020, Jürgen Hötzel <juergen at hoetzel.info>
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_restic;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_restic);
+#endif
+
+#include <string.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#include "aes.h"
+#include "arch.h"
+#include "base64_convert.h"
+#include "common.h"
+#include "formats.h"
+#include "misc.h"
+#include "poly1305-donna/poly1305-donna.h"
+#include "yescrypt/sysendian.h"
+#include "yescrypt/yescrypt.h"
+
+#define FORMAT_NAME "Restic Repository"
+#define FORMAT_LABEL "restic"
+#define FORMAT_TAG "$restic$"
+#define TAG_LENGTH sizeof(FORMAT_TAG) - 1
+#define ALGORITHM_NAME "PBKDF2/scrypt Poly1305"
+#define PLAINTEXT_LENGTH 125
+#define BINARY_SIZE 16 /* MAC */
+#define BINARY_ALIGN sizeof(uint32_t)
+#define DATA_SIZE 160
+#define SALT_ALIGN sizeof(uint32_t)
+#define MAC_SIZE 16
+#define NONCE_SIZE 16
+#define BENCHMARK_COMMENT ""
+#define BENCHMARK_LENGTH 7
+#define MIN_KEYS_PER_CRYPT 1
+#define MAX_KEYS_PER_CRYPT 1
+
+static int max_threads;
+static yescrypt_local_t *local;
+static char (*saved_key)[PLAINTEXT_LENGTH + 1];
+static uint32_t (*crypt_out)[BINARY_SIZE / sizeof(uint32_t)];
+static uint64_t poly1305_key_mask[2];
+
+typedef struct restic_salt {
+	int N;
+	int r;
+	int p;
```

Comment:
Now that you use `sizeof(uint32_t)` for `SALT_ALIGN`, I guess it's best to also use `uint32_t` for the salt struct fields. The `int` were because of `atoi()`, but our `isdec()` ensures a range that fits in `uint32_t` as well even if `int` were larger on some weird platform (in fact, a couple of platforms with 64-bit `int` existed historically).